ykllvmwrap/src/ykllvmwrap.cc

```diff
@@ -37,15 +52,107 @@ extern "C" char *__yk_llvmwrap_symbolizer_find_code_sym(
     return strdup(LineInfo->FunctionName.c_str());
 }
 
-/// Compiles an IRTrace to executable code in memory.
+// Load an LLVM module from an address.
+std::unique_ptr<Module> load_module(LLVMContext &Context, char *Ptr, size_t Len) {
+    auto Sf = StringRef(Ptr, Len);
+    auto Mb = MemoryBufferRef(Sf, "");
+    SMDiagnostic Error;
+    auto M = parseIR(Mb, Error, Context);
+    if (!M)
+        errx(EXIT_FAILURE, "Can't load module.");
+    return M;
+}
+
+// Compile a module in-memory and return a pointer to its function.
+extern "C" void *compile_module(Module *M) {
+    InitializeNativeTarget();
+    InitializeNativeTargetAsmPrinter();
+    InitializeNativeTargetAsmParser();
+
+    auto MPtr = std::unique_ptr<Module>(M);
+    ExecutionEngine *EE = EngineBuilder(std::move(MPtr)).create();
+	EE->finalizeObject();
+
+    if (EE->hasError())
+        errx(EXIT_FAILURE, "Couldn't compile trace: %s", EE->getErrorMessage().c_str());
+
+    return (void *) EE->getFunctionAddress(TRACE_FUNC_NAME);
+}
+
+// Compiles an IRTrace to executable code in memory.
 //
 // The trace to compile is passed in as two arrays of length Len. Then each
 // (FuncName[I], BBs[I]) pair identifies the LLVM block at position `I` in the
 // trace.
-extern "C" void __ykllvmwrap_irtrace_compile(char *FuncNames[], size_t BBs[], size_t Len) {
+//
+// Returns a pointer to the compiled function.
+extern "C" void *__ykllvmwrap_irtrace_compile(char *FuncNames[], size_t BBs[], size_t Len, char *SecPtr, size_t SecSize) {
+    LLVMContext Context;
+    auto DstMod = new Module("", Context);
+
+    // Set up new module for the trace.
+    llvm::FunctionType *FType = llvm::FunctionType::get(Type::getVoidTy(Context), false);
+	llvm::Function *DstFunc = llvm::Function::Create(FType, Function::InternalLinkage, TRACE_FUNC_NAME, DstMod);
```

Comment:
Odd indent here.

Reply:
And elsewhere.

Reply:
Yeah, noticed that. But probably not worth fixing manually if we run `clang-format` over it.

Reply:
Which reminds me, we also want to setup bors to check the formatting. Maybe we should do that before merging this PR.

Reply:
Let's leave it until after my branches are in to avoid merge conflicts?